tests/ui/manual_split_once.stderr

```diff
@@ -76,13 +76,13 @@ error: manual implementation of `rsplit_once`
   --> $DIR/manual_split_once.rs:41:13
    |
 LL |     let _ = "key=value".rsplitn(2, '=').next().unwrap();
-   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try this: `"key=value".rsplit_once('=').unwrap().1`
+   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try this: `"key=value".rsplit_once('=').map_or("key=value", |x| x.1)`
```

Comment:
> This suggestion doesn't seem like an improvement from the original code to me. I think it would make more sense to do `.rsplit('=').next().unwrap()`. But then it doesn't really fit with the `manual_split_once` lint and could be a new lint like...
> 
> > `needless_splitn`: lints usage of `splitn` or `rsplitn` where `split` or `rsplit` would be equivalent.

Hi, thanks for your patient review. According to your opinion, should I ignore this situation in this pr for this lint, and then develop a new lint and submit a new pr to support this situation.

Reply:
Sorry for the delay. Let's ignore this case for now and you can add a lint in a separate PR.

> In this example, I think rsplit will change the behavior of rsplitn. The result of rsplit depends on the string it processes.

Maybe I'm missing something. Do you have an example where `x.splitn(..).next()` is different from `x.split(..).next()`?

Reply:
Also for the new lint, I'd prefer the implementations to be together in one file named `str_split.rs` since they are closely related.

Reply:
> Sorry for the delay. Let's ignore this case for now and you can add a lint in a separate PR.
> 
> > In this example, I think rsplit will change the behavior of rsplitn. The result of rsplit depends on the string it processes.
> 
> Maybe I'm missing something. Do you have an example where `x.splitn(..).next()` is different from `x.split(..).next()`?

Hi, I can only think of this example at present:
let a1 = "key=value=add".splitn(1, '=').next().unwrap();
let a2 = "key=value=add".split('=').next().unwrap();
println!("{} {}", a1, a2);
result is : `key=value=add` and `key`
When pat >= 2 for splitn(), I think they are the same.

Reply:
> When pat >= 2 for splitn(), I think they are the same.

Exactly. I think that's where `needless_splitn` can lint.

Reply:
@rustbot label +S-waiting-on-author

Reply:
> Sorry for the delay. Let's ignore this case for now and you can add a lint in a separate PR.
> 
> > In this example, I think rsplit will change the behavior of rsplitn. The result of rsplit depends on the string it processes.
> 
> Maybe I'm missing something. Do you have an example where `x.splitn(..).next()` is different from `x.split(..).next()`?

Ignore the case of "key=value".rsplitn(2, '=').next().unwrap() for manual_split_once.